lib/src/map/map.dart

```diff
@@ -78,6 +78,41 @@ class MapControllerImpl implements MapController {
     return _state.rotate(degree, id: id, source: MapEventSource.mapController);
   }
 
+@override
+  LatLng? pointToLatLng(CustomPoint localPoint) {
+    if (_state.originalSize == null) {
+      return null;
+    }
+
+    final width = _state.originalSize!.x;
+    final height = _state.originalSize!.y;
+
+    final localPointCenterDistance =
+        CustomPoint((width / 2) - localPoint.x, (height / 2) - localPoint.y);
+    final mapCenter =
+        _state.options.crs.latLngToPoint(_state.center, _state.zoom);
+
+    var point = mapCenter - localPointCenterDistance;
+
+    if (_state.rotation != 0.0) {
+      point = rotatePoint(mapCenter, point);
+    }
+
+    return _state.options.crs.pointToLatLng(point, _state.zoom);
+  }
+
+  CustomPoint<num> rotatePoint(
+      CustomPoint<num> mapCenter, CustomPoint<num> point) {
+    final m = Matrix4.identity()
+      ..translate(mapCenter.x.toDouble(), mapCenter.y.toDouble())
+      ..rotateZ(-_state.rotationRad)
+      ..translate(-mapCenter.x.toDouble(), -mapCenter.y.toDouble());
+
+    final tp = MatrixUtils.transformPoint(
+        m, Offset(point.x.toDouble(), point.y.toDouble()));
+
+    return CustomPoint(tp.dx, tp.dy);
+  }
```

Comment:
Trying to appease the checker...

```suggestion
LatLng? pointToLatLng(CustomPoint localPoint) {
  if (_state.originalSize == null) {
    return null;
  }

  final width = _state.originalSize!.x;
  final height = _state.originalSize!.y;

  final localPointCenterDistance =
      CustomPoint((width / 2) - localPoint.x, (height / 2) - localPoint.y);
  final mapCenter =
      _state.options.crs.latLngToPoint(_state.center, _state.zoom);

  var point = mapCenter - localPointCenterDistance;

  if (_state.rotation != 0.0) {
    point = rotatePoint(mapCenter, point);
  }

  return _state.options.crs.pointToLatLng(point, _state.zoom);
}

CustomPoint<num> rotatePoint(
    CustomPoint<num> mapCenter, CustomPoint<num> point) {
  final m = Matrix4.identity()
    ..translate(mapCenter.x.toDouble(), mapCenter.y.toDouble())
    ..rotateZ(-_state.rotationRad)
    ..translate(-mapCenter.x.toDouble(), -mapCenter.y.toDouble());

  final tp = MatrixUtils.transformPoint(
      m, Offset(point.x.toDouble(), point.y.toDouble()));

  return CustomPoint(tp.dx, tp.dy);
}
```